### PR TITLE
Annotate implementations of IAliasSymbol

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Comment by sharwell** [comment]
@333fred can this be unblocked now?

**Comment by sharwell** [comment]
@dotnet/roslyn-compiler for review

File: src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs [hunk 1]
```diff
@@ -73,7 +76,9 @@ private AliasSymbol(Binder binder, SyntaxToken aliasName)
         }
 
         internal AliasSymbol(Binder binder, UsingDirectiveSyntax syntax)
+#nullable disable // Can 'syntax.Alias' be null here? https://github.com/dotnet/roslyn/issues/39166
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs]: No it cannot. There are only two call sites and they both validate that it's non-null.

File: src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs [hunk 2]
```diff
@@ -124,7 +124,7 @@ public override string Name
         {
             get
             {
-                return _aliasName.ValueText;
+                return _aliasName.ValueText ?? string.Empty;
```
Comment (sharwell) [inline src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs]: @jaredpar I'm guessing this is the expected behavior based on the recent change to `ValueText`

File: src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs (hunk 1 above)
Comment (sharwell) [reply src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs]: Resolved by passing the non-null arguments.

File: src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs (hunk 2 above)
Comment (jaredpar) [reply src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs]: Which recent changes are you referring to?

Comment (sharwell) [reply src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs]: `ValueText` was oblivious when I first made this PR. Now it returns a nullable string.

**Pull request merged**